lib/buildkite/test_collector/library_hooks/rspec.rb

```diff
@@ -29,6 +29,18 @@
     trace = Buildkite::TestCollector::RSpecPlugin::Trace.new(example, history: tracer.history)
     Buildkite::TestCollector.uploader.traces[example.id] = trace
   end
+
+  config.after(:suite) do
+    if Buildkite::TestCollector.artifact_path
+      filename = "#{Buildkite::TestCollector.artifact_path}/rspec-#{SecureRandom.uuid}.json.gz"
```

Comment:
[minor] This would make it slightly more redundant if someone includes a trailing slash in their artifact_path.

```suggestion
      filename = File.join(Buildkite::TestCollector.artifact_path, "rspec-#{SecureRandom.uuid}.json.gz")
```

Reply:
I think the filename should reference the collector rather than rspec. How about `buildkite-test_collector-#{SecureRandom.uuid}.json.gz`. Also, it would be great to include the build id rather than a uuid, but that might be more effort than it's worth.

Reply:
[minor] Maybe include a debug log that prints the filename created?

Reply:
Oh yeah good shout on the `File.join`. 

Yeah I think the file name should reference the collector too, and also have some sort of build/job level identifying ID in the name. The job/build UUIDs are available in the env, but the tricky bit here is that the artifacts need unique names, and we can't guarantee that a build, or a single job, would only contain one test run, so I've just opted for randomness for now. Basically I want to punt on a proper name for this file when it looks like it'll be getting wider release and not just something I'm mostly generating for test purposes